File: java-client/src/main/java/com/bolour/boardgame/client/domain/GameMiniState.java
/*
 * Copyright 2017 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */

package com.bolour.boardgame.client.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class GameMiniState {
    public final int lastPlayScore;
    public final int[] scores;
    public final boolean noMorePlays;

    @JsonCreator
    public GameMiniState(
      @JsonProperty("lastPlayScore") int lastPlayScore,
      @JsonProperty("scores") int[] scores,
      @JsonProperty("noMorePlays") boolean noMorePlays
    ) {
        this.lastPlayScore = lastPlayScore;
        this.scores = scores;
        this.noMorePlays = noMorePlays;
    }

}

File: haskell-server/src/BoardGame/Common/Domain/GameMiniState.hs
--
-- Copyright 2017-2018 Azad Bolour
-- Licensed under GNU Affero General Public License v3.0 -
--   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
--

{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE DeriveAnyClass #-}

module BoardGame.Common.Domain.GameMiniState (
  GameMiniState(..)
) where

import GHC.Generics (Generic)
import Control.DeepSeq (NFData)
import Data.Aeson (FromJSON, ToJSON)

data GameMiniState = GameMiniState {
    lastPlayScore :: Int
  , scores :: [Int]
  , noMorePlays :: Bool
}
  deriving (Eq, Show, Generic, NFData)

instance FromJSON GameMiniState
instance ToJSON GameMiniState

